main.c

```diff
@@ -128,7 +128,16 @@ static void linux_boot(const struct boot_params boot_params)
 
     printf("Loading Linux kernel from address: 0x%lx\n", boot_params.kernel_addr);
 
-    char *linux_header_end = (char *)(boot_params.kernel_addr + 0x201);
+    // Print relevant information about how the Linux kernel would like to be booted
+    const struct linux_params *const l_params = (const struct linux_params *const) boot_params.kernel_addr;
+    printf("Info from Linux Boot Protocol Header:\n");
+    printf("  kernel_alignment: 0x%x\n", l_params->kernel_alignment);
+    printf("  relocatable_kernel: %s\n", l_params->relocatable_kernel ? "true" : "false");
+    printf("  min_alignment: 0x%x\n", 0x1 << l_params->min_alignment);
+    printf("  pref_address: 0x%llx\n", l_params->pref_address);
+    printf("\n");
+
+    uint8_t *linux_header_end = (uint8_t *)(boot_params.kernel_addr + 0x201);
```

Comment:
Modifying the type isn't really a concern of this commit. Rather do that separately. It will make a `git blame` more useful. I suggest deferring that to the follow up MR.